_posts/2020-02-19-infrastructure-for-the-gem.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: "Setting up basic infrastructure for your gem"
+author: sheenaajay
+categories:
+  - gem
+  - development
+tags:
+  - walkthrough
+  - modules
+  - testing
+---
+
+## Introduction
+
+Here we will be going through the steps and best practices for setting up basic infrastructure for your gem. The steps given can be followed by anyone to setup the basic infrastructure for gem development which does the following.
+  - creates the basic file structure for the gem.
+  - adds the rubocop which validates the guidelines outlined in the community Ruby Style Guide.
+  - adds the travis or appveyor testing to the repo which validates the changes to the repo.
+  - adds the dependendabot to the repo which creates pull requests to keep dependencies up to date.
+
+> #### Note:
+>
+> This walkthrough assumes you have the following software installed:
+>
+> - [Ruby](https://www.ruby-lang.org/en/downloads/)
+>
+> It also assumes familiarity with the [Ruby gems](https://rubygems.org/).
+>
+
+## Infrastructure for your gem
+
+We're going through the steps and best practices for setting up basic infrastructure for your gem.
+
+For any repo created for developing a Gem we add owner, access and license to the repo. Follow the steps below to add them.
+
+1. Open a [git hub](https://github.com) page, perform the rest of these actions from there.
+   1. Click on + to create a new repo.
+      ![+]({% link /assets/2020-02-19-infrastructure-for-the-gem/newrepo.png %})
+
+   2. Provider the owner information, repo name and license.
+      ![owner information and repo name]({% link /assets/2020-02-19-infrastructure-for-the-gem/owner.png %})
+      ![license]({% link /assets/2020-02-19-infrastructure-for-the-gem/license.png %})
+
+   3. Click on Create repository button.
+
+   4. Open the new repo created and navigate to settings tab to update the access.
+
+   5. Provide the modules team as admin on github.
+      ![admin]({% link /assets/2020-02-19-infrastructure-for-the-gem/githubowner.png %})
```

Comment:
On a public post, we might want to tone this one down to "add *your* team as admin", so that we're not tagged on all the gems out there :->

Reply:
fixed this myself, realising what time it is.